cmd/juju/application/deploy.go

```diff
@@ -137,12 +138,29 @@ type modelConfigClient struct {
 	*modelconfig.Client
 }
 
+// cf charmrepo.Interface
+type charmrepoForDeploy interface {
+	Get(charmURL *charm.URL) (charm.Charm, error)
+	GetBundle(bundleURL *charm.URL) (charm.Bundle, error)
+	ResolveWithChannel(*charm.URL) (*charm.URL, params.Channel, []string, error)
+}
+
 type charmRepoClient struct {
-	*charmrepo.CharmStore
+	charmrepoForDeploy
+}
+
+type charmstoreForDeploy interface {
+	Get(endpoint string, extra interface{}) error
+	WithChannel(csparams.Channel) charmstoreForDeploy
+	// Latest(ids []*charm.URL, headers map[string][]string) ([]params.CharmRevision, error)
```

Comment:
If we don't need these methods, delete them.

Reply:
Done